fenix/app/src/androidTest/java/org/mozilla/fenix/ui/ComposeSearchTest.kt

```diff
@@ -611,4 +611,69 @@ class ComposeSearchTest {
             }
         }
     }
+
+    @Test
+    fun verifySearchTabsItemsTest() {
+        navigationToolbar {
+            verifyDefaultSearchEngine(engineName = "Google")
+            verifySearchSelectorArrowButton()
+            verifySearchBarPlaceholder(text = "Search or enter address")
+        }.clickUrlbar {
+            TestHelper.verifyKeyboardVisibility(isExpectedToBeVisible = true)
+            verifyScanButtonVisibility(visible = true)
+            verifyVoiceSearchButtonVisibility(enabled = true)
+            clickSearchSelectorButton()
+            selectTemporarySearchMethod("Tabs")
+            TestHelper.verifyKeyboardVisibility(isExpectedToBeVisible = true)
+            verifyScanButtonVisibility(visible = false)
+            verifyVoiceSearchButtonVisibility(enabled = true)
+            verifySearchBarPlaceholder(text = "Search tabs")
+        }
+    }
+
+    @Test
+    fun verifySearchTabsWithoutOpenTabsTest() {
+        navigationToolbar {
+        }.clickUrlbar {
+            clickSearchSelectorButton()
+            selectTemporarySearchMethod(searchEngineName = "Tabs")
+            typeSearch(searchTerm = "Mozilla")
+            verifyNoSuggestionsAreDisplayed(rule = activityTestRule, "Mozilla")
+            clickClearButton()
+            verifySearchBarPlaceholder("Search tabs")
+        }
+    }
+
+    @Test
+    fun verifySearchTabsWithOpenTabsTest() {
+        val firstPageUrl = TestAssetHelper.getGenericAsset(searchMockServer, 1)
+        val secondPageUrl = TestAssetHelper.getGenericAsset(searchMockServer, 2)
+
+        navigationToolbar {
+        }.enterURLAndEnterToBrowser(firstPageUrl.url) {
+        }.openComposeTabDrawer(activityTestRule) {
+        }.openNewTab {
+        }.submitQuery(secondPageUrl.url.toString()) {
```

Comment:
As discussed with Andi, he will try to mock the tab data using MockBrowserDataHelper.

Reply:
Made the changes @sv-ohorvath , successfully re-ran the UI tests 100x on Firebase.